app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryRepository.kt

```
@@ -0,0 +1,57 @@
+package org.simple.clinic.medicalhistory
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.simple.clinic.patient.SyncStatus
+import org.simple.clinic.patient.canBeOverriddenByServerCopy
+import org.simple.clinic.sync.SynceableRepository
+import java.util.UUID
+import javax.inject.Inject
+
+class MedicalHistoryRepository @Inject constructor(
+    val dao: MedicalHistory.RoomDao
+) : SynceableRepository<MedicalHistory, MedicalHistoryPayload> {
+
+  override fun recordsWithSyncStatus(syncStatus: SyncStatus): Single<List<MedicalHistory>> {
+    return dao.recordsWithSyncStatus(syncStatus).firstOrError()
+  }
+
+  override fun setSyncStatus(from: SyncStatus, to: SyncStatus): Completable {
+    return Completable.fromAction { dao.updateSyncStatus(from, to) }
+  }
+
+  override fun setSyncStatus(ids: List<UUID>, to: SyncStatus): Completable {
+    if (ids.isEmpty()) {
```

Question: Is it actually important to have this assertion? Either here, or in any of the other repositories? In the event that they are absent, the dao statement should just fail silently, right?

---

Just for the record, the dao throw an error if the arguments for an `IN` clause are empty.